src/patroni.py

```diff
@@ -185,6 +185,19 @@ def member_started(self) -> bool:
 
         return r.json()["state"] == "running"
 
+    @property
+    def is_database_running(self) -> bool:
```

Comment:
Yes, it timeouts - it is a clear sign that PostgreSQL doesn't work.
the goal of this test correctly detects situations like frozen processes because of some internal infinity loop or internal deadlock.

Reply:
I tested locally the two approaches and found some issues (maybe it's related to my environment). As Mykola commented on MM, I'll create a separate PR for this (ticket: https://warthogs.atlassian.net/browse/DPE-1690).